app/(app)/home.tsx

```diff
@@ -1,42 +1,77 @@
+import { Game } from '../../types/Game';
+import GameModal from '../../screens/GameModal';
 
 const auth = FIREBASE_AUTH;
 
-// create a component
+
 const HomePage = () => {
     const router = useRouter();
     const { games, loading: gamesLoading } = useGames();
     const [text, onChangeText] = React.useState('');
+    const [showGameModal, setShowGameModal] = useState(false);
+    const [game, setGame] = useState<Game>();
+
+    // eventually I think we will want a loading page here 
+    // if(gamesLoading){
+    //     return (
+    //         <ActivityIndicator
+    //             size="large"
+    //             color="#0000ff"
+    //         />)
+    // }
 
     return (
         <View style={[styles.container, styles.dark]}>
             <View style={styles.input}>
-                <MaterialCommunityIcons name="key-variant" size={20} color="white" />
-                <TextInput style={styles.textInput} value={text} placeholder={'Enter a code to join the game'}
-                    placeholderTextColor='white' onChangeText={onChangeText} />
+                <MaterialCommunityIcons
+                    name="key-variant"
+                    size={20}
+                    color="white"
+                />
+                <TextInput
+                    style={styles.textInput}
+                    value={text}
+                    placeholder={'Enter a code to join the game'}
+                    placeholderTextColor="white"
+                    onChangeText={onChangeText}
+                />
             </View>
-            <Pressable onPress={()=> console.log('Button Pressed')}
+            <Pressable
+                onPress={() => console.log('Button Pressed')}
                 style={({ pressed }) => [
                     styles.button,
-                    pressed && styles.buttonPressed,
-                  ]}
+                    pressed && styles.buttonPressed
+                ]}
             >
                 <Text style={styles.buttonText}>Join</Text>
             </Pressable>
-            <View style={{borderBottomColor: 'white', borderBottomWidth: 2, marginBottom: 5}}>
+            <View
+                style={{
+                    borderBottomColor: 'white',
+                    borderBottomWidth: 2,
+                    marginBottom: 5
+                }}
+            >
                 <Text style={styles.upcoming}>Upcoming Games</Text>
             </View>
             <View style={styles.games}>
                 {games.map((game, index) => (
-                    <GameCard key={index} game={game} />
+                    <GameCard
+                        key={index}
+                        game={game}
+                        onPress={() => {setGame(game); setShowGameModal(true);
+                        }}
+                    />
                 ))}
             </View>
+            {game && (<GameModal visible={showGameModal} onClose={() => setShowGameModal(false)} game={game}></GameModal>)}
         </View>
```

Comment:
this is the only real change in this file, the rest of the changes are prettier formatting changes